opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -594,15 +598,27 @@ def __init__(
             OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT,
         )
 
+        self.max_attribute_length = self._from_env_if_absent(
+            max_attribute_length,
+            OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT,
+        )
+        self.max_span_attribute_length = self._from_env_if_absent(
+            max_span_attribute_length,
+            OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT,
+            # use global attribute length limit as default
+            self.max_attribute_length,
+        )
```

Comment:
This block of code is confusing to me. I will pull this to local and check.

Reply:
Wow this whole limits stuff is confusing. Please check the L596-L610 once.

Reply:
Wow. My bad. I think it happened due to a bad rebase on github. Fixed now.